Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -72,6 +99,12 @@ public function getAggregateIdentifier(): string
      */
     public function recordThat(EventInterface $event, array $metadata = [])
     {
+        if ($this->initialized === false) {
+            // When calling self::recordThatMethod in the contructor the object is not initialized so event are queued
+            $this->pendingEvents[] = [$event, $metadata];
```

Comment:
`self::recordThatMethod`?

Will you call this method `recordThat` statically in your constructor?